numpy/core/tests/test_regression.py

```diff
@@ -2234,6 +2234,13 @@ def test_scalar_copy(self):
             item2 = copy.copy(item)
             assert_equal(item, item2)
 
+    def test_void_item_memview(self):
+        # gh-8129
+        va = np.zeros(10, 'V4')
+        x = va[:1].item()
+        del va
+        assert_equal(x, b'\x00\x00\x00\x00')
```

Comment:
Might be nice to set `va[0][0] = 1` above or something, to make this a slightly more thorough test.